src/bin/hexyl.rs

```diff
@@ -203,6 +203,15 @@ fn run() -> Result<()> {
                     width but still leave some space to the right.\nCannot be used with other \
                     width-setting options.",
                 ),
+        )
+        .arg(
+            Arg::new("little_endian_dump")
+                .short('e')
+                .action(ArgAction::SetTrue)
+                .help(
+                    "Print out the data in little endian format. Otherwise the data will be showed \
+                    in big endian format by default.",
+                ),
```

Comment:
Is this only usable with `--group-size`? If so, we should specify that using https://docs.rs/clap/latest/clap/struct.Arg.html#method.requires.

Also, should we maybe introduce a `--endianness=little/big` option instead (where we can still discuss what the default should be)? We could potentially still have `-e` as an alias for `--endianness=little` if we choose `big` as the default.

Reply:
I think this option doesn't have to be depending on `--group-size`. Because if we don't specify the group size in the option, we'll default it to one. In other words, `hexyl -g 1 file` and `hexyl file` are the same.


The reason that I choose to use big-endian as default is because of the design of `xxd`:
```
$ xxd --help
...
Options:
    -e          little-endian dump (incompatible with -ps,-i,-r).
```
However, it would be fine if `hexyl` want to have a different design for the dump of endianness. Please tell me which you would prefer.

Reply:
> I think this option doesn't have to be depending on --group-size. Because if we don't specify the group size in the option, we'll default it to one. In other words, hexyl -g 1 file and hexyl file are the same.

yes. But Endianness won't have any effect if the group size is 1, right?


> However, it would be fine if hexyl want to have a different design for the dump of endianness. Please tell me which you would prefer.

Let's follow xxd here for now, but I would still prefer a `--endianness=little/big` option instead of a `--little-endian-dump` flag.

Reply:
> But Endianness won't have any effect if the group size is 1, right?

Sure. If we want to inform the user that this would be no effect, let's use the [requires](https://docs.rs/clap/latest/clap/struct.Arg.html#method.requires) method.

> Let's follow xxd here for now, but I would still prefer a --endianness=little/big option instead of a --little-endian-dump flag.

I change the command line option to `--endianness=little/big` now.

Reply:
I thought about it again and now think that your initial version was actually better (sorry). Because *not* using `requires` allows someone to set an alias like `alias hexyl="hexyl --endianness=little"`, and use that even if `-g` is not present.

Instead, we can mention that it only has an effect if group size > 1. 

I'll make the necessary adjustments.